crates/oxc_parser/src/lexer/simd.rs

```diff
@@ -0,0 +1,86 @@
+//! Lexer methods using portable-SIMD
+//! See:
+//!   * <https://github.com/rust-lang/portable-simd/blob/master/beginners-guide.md>
+//!   * <https://rapidjson.org/md_doc_internals.html#SkipwhitespaceWithSIMD>
+//!   * <https://lemire.me/blog/2017/01/20/how-quickly-can-you-remove-spaces-from-a-string>
+
+use std::simd::{Simd, SimdPartialEq, ToBitMask};
+
+const ELEMENTS: usize = 16;
+type SimdVec = Simd<u8, ELEMENTS>;
+
+#[derive(Debug)]
+pub struct SkipWhitespace {
+    /// Total offset
+    pub offset: usize,
+
+    /// Found multiline comment end '*/'?
+    pub found: bool,
+
+    /// Found newline inside the comment?
+    pub newline: bool,
+
+    lf: SimdVec,
+    cr: SimdVec,
+    space: SimdVec,
+    tab: SimdVec,
+}
+
+impl SkipWhitespace {
+    pub fn new(newline: bool) -> Self {
+        Self {
+            offset: 0,
+            found: false,
+            newline,
+            lf: SimdVec::splat(b'\n'),
+            cr: SimdVec::splat(b'\r'),
+            space: SimdVec::splat(b' '),
+            tab: SimdVec::splat(b'\t'),
+        }
+    }
+
+    pub fn simd(mut self, bytes: &[u8]) -> Self {
+        let (chunks, remainder) = bytes.as_chunks::<ELEMENTS>();
+
+        for chunk in chunks {
+            self.check_chunk(chunk);
+            if self.found {
+                return self;
+            }
+        }
+
+        if !remainder.is_empty() {
+            // Align the last chunk for avoiding the use of a scalar version
+            let mut chunk = [0; ELEMENTS];
+            let len = remainder.len();
+            chunk[..len].copy_from_slice(remainder);
+            self.check_chunk(&chunk);
+        }
+
+        self
+    }
+
+    fn check_chunk(&mut self, chunk: &[u8]) {
+        let s = SimdVec::from_slice(chunk);
+
+        let any_newline = s.simd_eq(self.lf) | s.simd_eq(self.cr);
```

Comment:
This doesn't seem to handle non-ASCII newlines (U+2028 and U+2029). Is that handled elsewhere?

Reply:
These two are so rare so I left them in the scalar version

https://github.com/Boshen/oxc/blob/fe677d4909c49f6540a359218f00b773ed8bb23c/crates/oxc_parser/src/lexer/mod.rs#L452

Otherwise it'll drag down this simd version.